pkg/code-generator/codegen/generator.go

```diff
@@ -188,6 +191,19 @@ func generateKubeFilesForProject(project *model.Project) (code_generator.Files,
 			Content:  content,
 		})
 	}
+	for suffix, tmpl := range map[string]*template.Template{
+		"hack/update-codegen.sh": kube.GenerateScriptTemplate,
+		"hack/verify-codegen.sh": kube.VerifyGenerateTemplate,
+	} {
+		content, err := generateProjectFile(project, tmpl)
+		if err != nil {
+			return nil, errors.Wrapf(err, "internal error: processing template '%v' for project %v failed", tmpl.ParseName, project.ProjectConfig.Name)
+		}
+		v = append(v, code_generator.File{
+			Filename: filepath.Join("kube", suffix),
+			Content:  content,
+		})
+	}
```

Comment:
these are a LOT of changes to squeeze into a single PR. i understand the reasoning for wanting to do it together, but it makes reviewing and vetting each change significantly harder. Is there a good reason not to split them into a separate PR?
 
what is the reasoning behind generating these new files?

Reply:
We talked about this already. The old simple generation targets don't work because we can't reply on the GOPATH anymore. These scripts are based off of https://github.com/weaveworks/flagger/blob/master/hack/update-codegen.sh